### DISCUSSION TITLE
Is it possible to create a query that GROUPS BY with DATE_FORMAT without  using $queryRaw. See code

### DISCUSSION THREAD

**Discussion by Hugo-Persson** [prisma/prisma#20291] [Q&A] [answered]
I want to translate this code:
```ts
const emailsByMonth: [StatisticData] = await prisma.$queryRaw `SELECT COUNT(*) AS count, DATE_FORMAT(createdAt, '%Y-%m') AS period FROM SMTPLog WHERE createdAt > DATE_SUB(NOW(), INTERVAL 1 YEAR)  GROUP BY period`
```
To something using the built in `findMany` method. The problem is that I am grouping by `period` which is created with DATE_FORMAT.

**Comment by ludralph** [comment, marked as answer]
Hi @Hugo-Persson 👋 

Looking at the code you shared, I do not think it's possible to represent this using a `findMany` method in Prisma. The SQL functions `DATE_FORMAT` and `DATE_SUB` do no have a representation in Prisma. If you need to translate the given SQL query directly into a Prisma query without using raw database queries, the findMany method alone may not be sufficient. You would either need to use raw queries or perform additional data manipulation in JavaScript to achieve the same result.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by Hugo-Persson** [reply]
Yes that answers my question, I will probably use `rawQuery`. Thank you 🙏
